src/control/lib/control/system.go

```diff
@@ -251,7 +260,22 @@ func (resp *SystemQueryResp) UnmarshalJSON(data []byte) error {
 	if err := json.Unmarshal(data, &aux); err != nil {
 		return err
 	}
-	if err := resp.getAbsentHostsRanks(aux.AbsentHosts, aux.AbsentRanks); err != nil {
+	if err := sr.setAbsentHostsRanks(aux.AbsentHosts, aux.AbsentRanks); err != nil {
+		return err
+	}
+	return nil
+}
+
+// UnmarshalJSON unpacks JSON message into SystemQueryResp struct.
+func (resp *SystemQueryResp) UnmarshalJSON(data []byte) error {
+	type Alias SystemQueryResp
+	aux := &struct{ *Alias }{Alias: (*Alias)(resp)}
+
```

Comment:
Does all this aliasing still need to be done even though it is also being done at the `sysResponse` level?

Reply:
yes I tried for a while to extract it into something more elegant but this was the best I could get

Reply:
the alternative was repetitive boilerplate

Reply:
No worries then. Thanks.